pyproject.toml

```diff
@@ -78,20 +78,25 @@ dependencies = [
   # Schema validation
   "jsonschema",
 
-  # Preview
-  "canals==0.8.0",
-  "openai",
-  "Jinja2",
-  "openai-whisper",  # FIXME https://github.com/deepset-ai/haystack/issues/5731
-  "pypdf",
-
   # Agent events
   "events",
 
   "requests-cache<1.0.0",
 ]
 
 [project.optional-dependencies]
+preview = [
+  "canals==0.8.0",
+  "requests",
+  "pandas",
+  "rank_bm25",
+  "tqdm",
+  "tenacity",
+  "lazy-imports",
+
+  "Jinja2",
+  "openai",
```

Comment:
I am on the fence regarding these two. Clearly not all components need these, so they should be optional. However, they're very often needed in our most prominent usecase (RAG) and relatively lightweight. Wdyt?

Reply:
Okay, Stefano and I agree that we can keep them in here. This way, together with the rank_bm25 dependency, we enable RAG out of the box with open ai models.

Reply:
I would include them, they're lightweight and well maintained. If we keep them, why the optional import?

Reply:
In case of need a power user can still import Haystack without them if needed. If we remove the optional import that's not possible anymore. However that's quite arbitrary, I'm happy to remove it if it's more tidy.

Reply:
Conditional imports removed